packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -0,0 +1,344 @@
+   *
+   * @param options The object containing all the options for the credential request
+   * @param credentialRecord the credential record containing the offer from which this request
+   * is derived
+   * @returns object containing associated attachment, formats and requestAttach elements
+   *
+   */
+  public async createRequest(
+    options: FormatServiceRequestCredentialOptions,
+    credentialRecord: CredentialExchangeRecord
+  ): Promise<FormatServiceCredentialAttachmentFormats> {
+    if (!options.offerAttachment) {
+      throw new AriesFrameworkError(
+        `Missing attachment from offer message, credential record id = ${credentialRecord.id}`
+      )
+    }
+    const formats: CredentialFormatSpec = {
+      attachId: uuid(),
+      format: 'aries/ld-proof-vc-detail@v1.0',
+    }
+
+    // W3C message exchange can begin with request or there could be an offer.
+    // Use offer attachment as the credential if present
+    // otherwise use the credential format payload passed in the options object
+
+    const credOffer = options.offerAttachment.getDataAsJson<SignCredentialOptions>()
+    const attachment = credOffer ? credOffer : options.jsonld?.credentialSubject
+
+    const requestAttach: Attachment = this.getFormatData(attachment, formats.attachId)
+
+    return { format: formats, attachment: requestAttach }
+  }
+
+  /**
+   * Create a {@link AttachmentFormats} object dependent on the message type.
+   *
+   * @param options The object containing all the options for the credential to be issued
+   * @param record the credential record containing the offer from which this request
+   * is derived
+   * @param requestAttachment the attachment containing the request
+   * @returns object containing associated attachment, formats and requestAttach elements
+   *
+   */
+  public async createCredential(
+    options: ServiceAcceptRequestOptions,
+    record: CredentialExchangeRecord,
+    requestAttachment: Attachment
+  ): Promise<FormatServiceCredentialAttachmentFormats> {
+    if (!requestAttachment || !requestAttachment?.data?.base64) {
+      throw new AriesFrameworkError(
+        `Missing request attachment from request message, credential record id = ${record.id}`
+      )
+    }
+
+    const formats: CredentialFormatSpec = {
+      attachId: uuid(),
+      format: 'aries/ld-proof-vc@1.0',
+    }
+
+    const attachmentId = options.attachId ? options.attachId : formats.attachId
+
+    // sign credential here. credential to be signed is received as the request attachment
+    // (attachment in the request message from holder to issuer)
+    const credentialOptions = requestAttachment?.getDataAsJson<SignCredentialOptions>()
+    const signCredentialOptions: SignCredentialOptions = {
+      credential: JsonTransformer.fromJSON(credentialOptions.credential, W3cCredential),
+      proofType: credentialOptions.proofType,
+      verificationMethod: credentialOptions.verificationMethod,
+    }
```

Comment:
(same answer as in the other comment)

The properties you mentioned are part of the W3cCredentialService API as optional properties as defined in the spec. Please see the interface [here](https://github.com/hyperledger/aries-framework-javascript/blob/be57da5725fe10802fc6cd12a75bf963c4a839fd/packages/core/src/modules/vc/models/W3cCredentialServiceOptions.ts#L9).

I could throw an error in the `sign` function of the `W3cCredentialService` if `credentialStatus` is provided, but I feel this is more appropriate at a higher level since that is tightly coupled to the protocol itself. Any thoughts on this @TimoGlastra?